### DISCUSSION TITLE
Decoding data file

### DISCUSSION THREAD

**Discussion by shlomibr** [opendcs/opendcs#214] [Development Practices]
Hi,

I got a request to decode data file retrieved from NOAA. The format is encoded and I didnt find how to decode it.
Does someone know how to deal with it?

example of a single row - 34A0701023074035040G37+0NN028WUB00029bB1|@FeCUAATdATdATdATd@t~@SQ 

Thanks,
Shlomi

**Comment by mikecovesw** [comment]
According to the PDT, 34A07010 is a Bureau of Reclamation site. National Weather Service HADS is a good resource:
https://hads.ncep.noaa.gov/
Key in the DCP address and it will get you to a page with a list of sensors and the most recent data.

**Reply by shlomibr** [reply]
Thanks for the response but how do I decode the rest of the string? it should be decoded to timeserias data.

**Reply by MikeNeilson** [reply]
you'll need to create a "configuration" with a script and attach it to a platform. The DECODING portion of the manual is here: https://opendcs-env.readthedocs.io/en/latest/decoding-guide.html, specifically for that data you'll need pseudo binary. 

Decoding the data may involve a bit of trial and error as there are several options for pseudo binary. 

Alternatively, knowing that it's USBR data you may be able to contact the USBR office nearest the site and get the correct information from them. No guarantees, but worth asking.

**Reply by shlomibr** [reply]
Thank you !!!
